docs/software/container-engine/guidelines-images/image-nccl-tests.md

```diff
@@ -0,0 +1,183 @@
+[](){#ref-ce-guidelines-images-nccl-tests}
+# NCCL Tests image
+
+This page describes a container image featuring the [NCCL Tests](https://github.com/NVIDIA/nccl-tests) to demonstrate how to efficiently execute NCCL-based containerized software on Alps.
+
+This image is based on the [OpenMPI image][ref-ce-guidelines-images-ompi], and thus it is suited for hosts with NVIDIA GPUs, like Alps GH200 nodes.
+
+A build of this image is currently hosted on the [Quay.io](https://quay.io/) registry at the following reference:
+`quay.io/ethcscs/nccl-tests:2.17.1-ompi5.0.8-ofi1.22-cuda12.8`.
+
+## Contents
+
+- Ubuntu 24.04
+- CUDA 12.8.1 (includes NCCL)
+- GDRCopy 2.5.1
+- Libfabric 1.22.0
+- UCX 1.19.0
+- OpenMPI 5.0.8
+- NCCL Tests 2.17.1
+
+## Containerfile
+```Dockerfile
+FROM quay.io/ethcscs/ompi:5.0.8-ofi1.22-cuda12.8
+
+ARG nccl_tests_version=2.17.1
+RUN wget -O nccl-tests-${nccl_tests_version}.tar.gz https://github.com/NVIDIA/nccl-tests/archive/refs/tags/v${nccl_tests_version}.tar.gz \
+    && tar xf nccl-tests-${nccl_tests_version}.tar.gz \
+    && cd nccl-tests-${nccl_tests_version} \
+    && MPI=1 make -j$(nproc) \
+    && cd .. \
+    && rm -rf nccl-tests-${nccl_tests_version}.tar.gz
+```
+
+!!! note
+    This image builds NCCL tests with MPI support enabled.
+
+## Performance examples
+
+### Environment Definition File
+```toml
+image = "quay.io#ethcscs/nccl-tests:2.17.1-ompi5.0.8-ofi1.22-cuda12.8"
+
+[env]
+PMIX_MCA_psec="native"
+
+[annotations]
+com.hooks.aws_ofi_nccl.enabled = "true"
+com.hooks.aws_ofi_nccl.variant = "cuda12"
+```
+
+### Notes
+
+- Since OpenMPI uses PMIx for wire-up and communication between ranks, when using this image the `srun` option `--mpi=pmix` must be used to run successful multi-rank jobs.
+- NCCL requires the presence of the [AWS OFI NCCL plugin](https://github.com/aws/aws-ofi-nccl) in order to correctly interface with Libfabric and (through the latter) the Slingshot interconnect. Therefore, for optimal performance the [related CE hook][ref-ce-aws-ofi-hook] must be enabled and set to match the CUDA version in the container.
```

Comment:
This could be a warning box, since failing to do this will result in poor performance.